src/sif/bicyclecost.cc

```diff
@@ -575,27 +564,12 @@ bool BicycleCost::AllowedReverse(const baldr::DirectedEdge* edge,
     return false;
   }
 
-  if (edge->access_restriction()) {
-    const std::vector<baldr::AccessRestriction>& restrictions =
-        tile->GetAccessRestrictions(opp_edgeid.id(), kBicycleAccess);
-    for (const auto& restriction : restrictions) {
-      if (restriction.type() == AccessType::kTimedAllowed) {
-        // allowed at this range or allowed all the time
-        return (current_time && restriction.value())
-                   ? (edge->surface() <= worst_allowed_surface_ &&
-                      IsRestricted(restriction.value(), current_time, tz_index))
-                   : true;
-      } else if (restriction.type() == AccessType::kTimedDenied) {
-        // not allowed at this range or restricted all the time
-        return (current_time && restriction.value())
-                   ? (edge->surface() <= worst_allowed_surface_ &&
-                      !IsRestricted(restriction.value(), current_time, tz_index))
-                   : false;
-      }
-    }
-  }
   // Prohibit certain roads based on surface type and bicycle type
-  return opp_edge->surface() <= worst_allowed_surface_;
+  if (edge->surface() > worst_allowed_surface_) {
+    return false;
```

Comment:
can we remove the branch here and add this to the return statement. it might be less clear but we call this method a lot.

Reply:
I'll see if I can benchmark this. I was hoping a modern compiler would output pretty equivalent code for the suggested change and current code.

Reply:
I updated the PR description with benchmarks. Not sure any difference is measurable, what do you think?